src/main/java/org/pesmypetcare/webservice/dao/MedicationDaoImpl.java

```diff
@@ -0,0 +1,176 @@
+            throws DatabaseAccessException {
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        DocumentReference medicationsRefDoc = medicationsRef.document(dateName);
+        ApiFuture<DocumentSnapshot> future = medicationsRefDoc.get();
+        DocumentSnapshot medicationDoc;
+        try {
+            medicationDoc = future.get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+        if (!medicationDoc.exists()) {
+            throw new DatabaseAccessException(INVALID_MEDICATION_EXC, MEDICATION_DOES_NOT_EXIST_EXC);
+        }
+        return medicationDoc.toObject(MedicationEntity.class);
+    }
+
+    @Override
+    public List<Map<List<String>, Object>> getAllMedicationData(String owner, String petName)
+            throws ExecutionException, InterruptedException {
+        List<String> pks = new ArrayList<>();
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        List<Map<List<String>, Object>> externalList = new ArrayList<>();
+        String aux;
+        ApiFuture<QuerySnapshot> future = medicationsRef.get();
+        List<QueryDocumentSnapshot> medicationDocuments = null;
+            medicationDocuments = future.get().getDocuments();
+        for (QueryDocumentSnapshot medicationDocument : medicationDocuments) {
+            Map<List<String>, Object> internalList = new HashMap<>();
+            aux = medicationDocument.getId();
+            pks.add(pkToDate(aux));
+            pks.add(pkToName(aux));
+            internalList.put(Collections.singletonList(DATENAME), pks);
+            pks.clear();
+            internalList.put(Collections.singletonList(BODY),
+                    medicationDocument.toObject(MedicationEntity.class));
+            externalList.add(internalList);
+        }
+        return externalList;
+    }
+
+    @Override
+    public List<Map<List<String>, Object>> getAllMedicationsBetween(String owner, String petName,
+                                                                    String initialDate, String finalDate)
+            throws ExecutionException, InterruptedException {
+        List<String> pks = new ArrayList<>();
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        List<Map<List<String>, Object>> externalList = new ArrayList<>();
+        String aux;
+        ApiFuture<QuerySnapshot> future = medicationsRef.get();
+        List<QueryDocumentSnapshot> medicationDocuments;
+        medicationDocuments = future.get().getDocuments();
+        for (QueryDocumentSnapshot medicationDocument : medicationDocuments) {
+            Map<List<String>, Object> internalList = new HashMap<>();
+            aux = medicationDocument.getId();
+            if (initialDate.compareTo(pkToDate(aux)) < 0 && finalDate.compareTo(pkToDate(aux)) > 0){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['{' is not preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)